### PR TITLE
#ME-561 | add support for module reset

### PR DESCRIPTION
**Pull request by Niryo**

### CODE REVIEW

File: src/jest-setup.ts [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-
-// make sure that axios is a singleton
-let actualAxios: any;
-jest.doMock('axios', () => {
-  if (!actualAxios) {
-    actualAxios = jest.requireActual('axios');
-  }
-  return actualAxios;
-}, { virtual: true });
+require('axios').defaults.adapter = require('./axios-fetch-adapter').default;
```
Comment (doronpr) [inline src/jest-setup.ts:1]: I don't want to ask why this is outside of the beforeEach :)

File: tests/settings.spec.ts [hunk 2]
```diff
@@ -9,23 +9,31 @@ describe('Settings', () => {
 
   it('should make a real network call if network is enabled and an unmocked request is fetched', async () => {
     require('netmock-js').allowRealNetwork(true);
-    await fetch('https://wix.com');
-    await axios.get('https://wix.com');
-    await expect((global as any).fetchSpy).toHaveBeenNthCalledWith(1, 'https://wix.com', undefined);
-    await expect((global as any).fetchSpy).toHaveBeenCalledTimes(2);
+    await expect(async () => {
+      await fetch('https://this-site-does-not-exist-niryo.com');
+    }).rejects.toThrowError('ENOTFOUND');
```
Comment (doronpr) [inline tests/settings.spec.ts:14]:
```suggestion
    await expect(fetch('https://this-site-does-not-exist-niryo.com')).rejects.toThrowError('ENOTFOUND');
```
wouldn't this be equivilent?

File: tests/settings.spec.ts [hunk 3]
```diff
@@ -9,23 +9,31 @@ describe('Settings', () => {
...
+
+    await expect(async () => {
```
Comment (doronpr) [inline tests/settings.spec.ts:16]: same as above

File: tests/settings.spec.ts [hunk 4]
```diff
@@ -9,23 +9,31 @@ describe('Settings', () => {
...
-    try {
-      await fetch('https://blamos.com');
-      await axios.get('https://blamos.com');
-    } catch {
-      // do nothing
-    }
-    await expect((global as any).fetchSpy).toHaveBeenCalledWith('https://wix.com', undefined);
-    await expect((global as any).fetchSpy).toHaveBeenCalledTimes(2);
+    require('netmock-js').allowRealNetwork(/this-site-does-not-exist-niryo/);
+    await expect(async () => {
```
Comment (doronpr) [inline tests/settings.spec.ts:23]: all these are same as above

**Review by doronpr — APPROVED** [review]

File: src/jest-setup.ts (hunk 1 above)
Comment (Niryo) [reply src/jest-setup.ts:1]: lol, yea...I have now clue sadly. it just doesn't work from within, I think it happens too late when it's in the beforeEach.

File: tests/settings.spec.ts (hunk 2 above)
Comment (Niryo) [reply tests/settings.spec.ts:14]: checking

Comment (Niryo) [reply tests/settings.spec.ts:14]: you are right, it works. pushing a fix now

**Pull request merged**